psc/src/main/java/com/pinterest/psc/metadata/client/PscMetadataClient.java

```diff
@@ -0,0 +1,172 @@
+    }
+
+    /**
+     * Describe the metadata for the given {@link TopicRn}'s in the cluster.
+     *
+     * @param clusterUri
+     * @param topicRns
+     * @param timeout
+     * @param timeUnit
+     * @return a map of {@link TopicRn} to {@link TopicRnMetadata}
+     * @throws ExecutionException
+     * @throws InterruptedException
+     * @throws TimeoutException
+     */
+    public Map<TopicRn, TopicRnMetadata> describeTopicRns(TopicUri clusterUri, Set<TopicRn> topicRns, long timeout, TimeUnit timeUnit) throws ExecutionException, InterruptedException, TimeoutException {
+        PscBackendMetadataClient backendMetadataClient = getBackendMetadataClient(clusterUri);
+        return backendMetadataClient.describeTopicRns(topicRns, timeout, timeUnit);
+    }
+
+    /**
+     * List the offsets for the given {@link TopicUriPartition}'s in the cluster.
+     *
+     * For each {@link TopicUriPartition}, the user can specify an {@link MetadataClientOption} to specify whether to
+     * get the earliest or latest offset.
+     *
+     * For example, if the user specifies {@link MetadataClientOption#OFFSET_SPEC_EARLIEST}, the client will return the earliest
+     * offset for the partition.
+     *
+     * @param clusterUri
+     * @param topicRnsAndOptions
+     * @param timeout
+     * @param timeUnit
+     * @return a map of {@link TopicUriPartition} to {@link MessageId}. The {@link MessageId} will contain the offset but
+     * not necessarily the timestamp (timestamp might be null or unset)
+     * @throws ExecutionException
+     * @throws InterruptedException
+     * @throws TimeoutException
+     */
+    public Map<TopicUriPartition, MessageId> listOffsets(TopicUri clusterUri, Map<TopicUriPartition, PscMetadataClient.MetadataClientOption> topicRnsAndOptions, long timeout, TimeUnit timeUnit) throws ExecutionException, InterruptedException, TimeoutException {
+        PscBackendMetadataClient backendMetadataClient = getBackendMetadataClient(clusterUri);
+        return backendMetadataClient.listOffsets(topicRnsAndOptions, timeout, timeUnit);
+    }
+
+    /**
+     * List the offsets for the given {@link TopicUriPartition}'s in the cluster for the given consumer group ID.
+     *
+     * @param clusterUri
+     * @param consumerGroup
+     * @param topicUriPartitions
+     * @param timeout
+     * @param timeUnit
+     * @return a map of {@link TopicUriPartition} to {@link MessageId}. The {@link MessageId} will contain the offset but
+     * not necessarily the timestamp (timestamp might be null or unset)
+     * @throws ExecutionException
+     * @throws InterruptedException
+     * @throws TimeoutException
+     */
+    public Map<TopicUriPartition, MessageId> listOffsetsForConsumerGroup(TopicUri clusterUri, String consumerGroup, Collection<TopicUriPartition> topicUriPartitions, long timeout, TimeUnit timeUnit) throws ExecutionException, InterruptedException, TimeoutException {
```

Comment:
This might require a wider change. Instead of `long timeout, TimeUnit timeUnit` can we use `java.time.Duration`?